src/components/Footer/JoinUs/JoinUs.js

```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react';
+import './JoinUs.scss';
+import '../Footer.scss';
+
+class JoinUs extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: '',
+      formError: {
+        email: '',
+      },
+      emailValid: false,
+      formValid: false,
+    };
+  }
+
+  handleInput(e) {
+    e.preventDefault();
+    const name = e.target.name;
+    const value = e.target.value;
+    this.setState({ [name]: value }, () => {
+      this.validateField(value);
+    });
+  }
+
+  validateField(value) {
+    const emailRegEx = /^(\D)+(\w)*((\.(\w)+)?)+@(\D)+(\w)*((\.(\D)+(\w)*)+)?(\.)[a-z]{2,}$/;
```

Comment:
regex should be in constant variable in the top of the file or even in constant file/folder